File: activity/app/src/main/java/com/rikkei/training/activity/activity/CActivity.java
package com.rikkei.training.activity.activity;


import android.content.Intent;
import android.view.View;
import android.widget.Button;

import com.rikkei.training.activity.R;
import com.rikkei.training.activity.base.BaseActivity;

public class CActivity extends BaseActivity {
    private Button mButtonC;
    private Button mButtonAResult;

    @Override
    protected void initData() {

    }

    @Override
    protected void initViews() {
        mButtonC = findViewById(R.id.btn_act_cd);
        mButtonC.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                Intent intent = new Intent(CActivity.this,DActivity.class);
                startActivity(intent);
            }
        });
        mButtonAResult = findViewById(R.id.btn_finish);
        mButtonAResult.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                Intent intent = new Intent();
                setResult(RESULT_OK,intent);
                finish();
            }
        });
    }

    @Override
    protected int getLayoutId() {
        return R.layout.activity_c;
    }
}

File: Activitykotlin/app/src/main/java/com/rikkei/training/activitykotlin/Activity/CActivity.kt
package com.rikkei.training.activitykotlin.Activity

import android.app.Activity
import android.content.Intent
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.view.View
import com.rikkei.training.activitykotlin.R
import kotlinx.android.synthetic.main.activity_c.*

class CActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_c)
        btn_gt_cd.setOnClickListener(View.OnClickListener {
            intent = Intent(this, DActivity::class.java)
            startActivity(intent)
        })
        btn_finish.setOnClickListener(
            View.OnClickListener {
                intent = Intent()
                setResult(Activity.RESULT_OK, intent)
                finish()
            }
        )

    }
}
